### ISSUE TITLE
when sniff a response used gzip, decompress fail

### ISSUE THREAD

**Issue by Qnner** [secdev/scapy#2744]
#### Environment

-   Scapy version: `scapy version 2.3.4`
-   Python version: `2.7`
-   Operating System: `Windows and Centos 7.6`

#### How to reproduce

when I sniff a http response which's header, Content_Encoding is gzip, I can not get the json file correctly，
I noticed that
 "Perform decompression / defragmentation
on all TCP streams simultaneously, but
only acts passively." in https://scapy.readthedocs.io/en/latest/layers/http.html

and "This will decode HTTP packets using Content_Length or chunks, and will also decompress the packets when needed. Note: on failure, decompression will be ignored." in https://scapy.readthedocs.io/en/latest/api/scapy.layers.http.html

but I still cannot decompress my response.

t tried add a new line in scapy.layer.http.py(near line 288 in function 'post_dissect'):
like this:
```python
        # Decompress
        try:
            if "deflate" in encodings:
                import zlib
                s = zlib.decompress(s)
            elif "gzip" in encodings:
                print "======> hit gzip"
                s = gzip_decompress(s)
            elif "compress" in encodings:
                import lzw
                s = lzw.decompress(s)
        except Exception as e:
            print '===> hit Exception'
            print e
            # Cannot decompress - probably incomplete data
            pass
        return s
```

and I am sure now that my code run this Decompress yet, but not success.

here is my original code：

```python
def packet_parse(pkt):
    if pkt.haslayer(HTTPRequest):
        return
    elif pkt.haslayer(HTTPResponse):
        print pkt[HTTPResponse]



if __name__ == '__main__':
    load_layer("http")
    conf.contribs["http"]["auto_compression"] = True
    sniff(offline=r'C:\Users\test1\Downloads\http-chunked-gzip.pcap', prn=packet_parse, session=TCPSession)
```

#### Actual result

```
HTTP/1.1 200 OK
Server: cloudflare-nginx
Date: Wed, 06 Jan 2016 20:42:10 GMT
Content-Type: text/html
Transfer-Encoding: chunked
Connection: close
Set-Cookie: __cfduid=d8d37b52eaa3137bdfd7fd67a4ffc8a7a1452112929; expires=Thu, 05-Jan-17 20:42:09 GMT; path=/; domain=.wireshark.org; HttpOnly
X-Frame-Options: SAMEORIGIN
Strict-Transport-Security: max-age=31536000;
X-Slogan: It's a great product with a great story to tell. I'm pumped!
X-Mod-Pagespeed: 1.9.32.11-7550
Vary: Accept-Encoding
Cache-control: max-age=0, no-cache, no-store
X-Slogan: Go deep.
CF-RAY: 260a3f709d7b0761-AMS
Content-Encoding: gzip

         ��     106c
�x��}�}������M�I6�i�aw��bU�!/ �(�����t���sHl*z;�����slB�3��ݿ��^���_�4�{7a���w�2m�^�i�zW�Θ�ݽ='����O_tNN�=??k�Ο��4�e�a��i�E�	�Z����W��*�h�P�M�L���V��]������(���J�RM�S.#5uDw�(:�����X�����ֲ0)�NԍT���h��R٦c�M�X�tTw���uw�Jf,ƄG���{v�o�D3��6�sߍ�P#�x.��w͞3�F^���������,�]����?���Pb��Ny"�n�]��Hy����9��%:��G��]џ8���G��s$Y������|��7��nY�Q�u�I�?�e\��6�Z�����Y��j8��˸���,�3hWF����˰�>����M������/��&�jjP�h5�y�wo�G3/c��Vz\��GJ��3�;�]��N좓8#&��H��5��{��vӽw)y��:Iݷ�X�S��{�{G�w ��*�w�S�k3���{��Ei��<˗n����4E� L�6h�?_����c˭��1az��7�Rp��y�~ٸs,��Fэ�q3�#�a��Q���#%��b�b�,�����ݳ�o�3�Y�JcD5��$s��_;��I���0��6�-�P�F�ZOi��#6�V���v�a�Q��3��}��ٔG6�F8�!�ŋ\r˙pM�v�^���؝p�fJۇ s4e�e�T\�;��eY�,c��{��4	�}��M0E��j��1��r&���K/4�)F�uM�e�X@47��|3b)��@�:įL��n�:�Z�~��|�j9O[-�Y��(i�2�Cj����O͇
J��K�c����/����:\�J�IrO�����ڨ�u&�F�Xy���B9ӯ��kgF�Й5�	�mо�3��Io�X�����*Q�+��\+لǌ����q.V�S3�<j'�ܯ���VY��kvz"qj�k��J@�?�����hX�B���qi1�ґŔ�6h�����Ch���Q���Y����l)�B��x)ʼ��Tn��^��u��j7�2��;���t����~ T�-��)��މP�
��zl��s�~���h�^��f�v�Th|2yY��b
��y����^�Hu�s;�����l��=b]����v� �����I�r��YnU�,F��%�h,��\D��8���F�|�:��H1��\� S�(�kW�
��%[��X=K6YnN�O��T��&Ӗ�e��iN�#կ�i0TD�*2�$,R��S�RuTg,����y+�5�r��t���j�z�1B��ns����].�V�}7��`Bc���T��Zk9�B5A�F/O�h�;��a�֚)�I�C8h}qT�P�A��n�/�g��Y��@7P֪t;ó��uGV�Ya ��C����}�J`�mN �����J�i�;P�8DiQ7�cY�u7,]˄(rښP�ƥvt��\晥�T�Bn��X���n�٩nm����*le\&yZ�K����`�2�{�����^kD���kJV�J�C��x��/��V鼖y�n5�@�����Sib��|�aV-ޚH�3P��(O��"\�ʨ�^��v�>[&a�mU���D�rZu��K̈����l���r줽H�K�%:B��פr`�R��&��������������3.��20>��2-��ŁR�O��Kh�o��j*0���[5�4���{��|�����.m�*�b��>~���Jn��w�&�ʶ`�a"�P�|���Ź�z�Se.+����U��AřM�m��w��P��κiE��n���QW�5K<�QC¸.��N�c���a������v$V�U�t�e%�!t��fa��(e�O���f��C�z���pW�������K�[q�Ӗ}��a� ��66�����>�c��ҟ�:���OpY�<i�x�9�����v��X?��,X�l��V�;�ӧ��_"��F�1��N�M ��<�a�td��a��6>�Z��������6Q�l9�yҦ�G^+�֪��{,��zh�67c�\\Bi�-
�{�,򧚾O*E|4B�)����
 �2k��Y���
`M_<GZ��p��������mnoo�o�.O��89Z��*��
E�5�1�)�"���ʁ�gg�N�:�a�2Zݩ�Y���C=�ʜ=�(/�0-��a]������u�{*-�2��n'����r����m)��m߿��wV,L��0qi���1�7�j�o#�L?�<�(6�jl��7��r[���)��t�V
�2'���p�Wi|B,�����cq�f�i���I\���Q(�F�o�����(LօL�6�m�m��#|˄X�O��'aB�U=�v��
fČ�gᘀ�k\�7�񈣆�]2ZC�?   ��
e1e
���Gs��^9+T��h�t��ɲI�[�
az�+v�ՠLХ%�m�=��m�����=Pɏ����x���s5�����c������b羷�x;005C pM�Fipֈ;�5v�.��E(���Z:�욦�^�
����I6�Ϋ�JKU����v��}��dHb�x;�c@Wp�' �I<a�Y��	����GW��vT^�F|�y�	�^��+��]ك]�4$�D����
;��
q�"h���l�L�}�4Q�9@�Kx��/�V�tS�u�
d'XE����Xx�Ω n��M�V&!�C$���]:P�x�-fض&EJY=~�(�w
;�"�u#Lvo8���y�0�������\%�r�iy>�87ď>����nِ;�DJUt��`��O��m;]G3\wm��J�n���q
�gو�~F��bN��+j����&9�S�Һ���Ɩ�X l�~�=��{_��g�-��n����c:����jZ
��\T�J<�5�w2�R%okP�um��w0���n�$LuM]<aĵ�|p�wJ��=�Vq=[c�%�ԗ�2�&�7��1Py�V��F.��ā�{,����\~�~�~Zʻ�9���銅7�ܻ�
�_�7��*��r��wV`��)��4LO�������J�v��e�A�aCs���@���7�N��6�%�9�UQ"�O��Sm�)�q_8ߪq�5=�us��%�85���S��Ϫ*��B�4��lm�F���Q��nЅl1V�y�Q��ލĉ�,m�4b	U���j�-V��+���(n+�����0L��萁E�J*��X�6J��H:�wqF�o�-�=-����b΅1��t�1Mc5�k��5��xD��nU=EPI+jE=(8�%�uؼ��I��F���죱q�p�$�'�d������B٩5�J� �����z_�n��Y����fk�V֨u��$�HSZE��7�;��>@�K�sΩ�<tf��Y��-d��<�����\��A�g��hަ_�N�t`��k[�6�uӖ�'E
����SWoA=Kɩ��&M�XSs�D����0���.�B���Ԯ�$�57i�h6}��g��w�X�����V�)��Z��p��p�����D֙���cR��x�B�9�lq>�y�
L�9�/F/���m�{��<��ĕ�uO�j��>!�ty3�k�}��En�k�ST��,5�{]�=���w��n���ٮ�>�8�tO��A������Y��c̼vՎ�3\i�P��h��
a����o�*5MuP?{e!�
�xڵ-���?���NXwZ����� �4���#��l憋��z����f��J��e��g��M^(��Q|�c7�$9�qm;�6�{g%:�.of�:��nsg�z��)
��)��|�q���U���8g1����tO���J[�`l�+m�\���vg(4�~�8ao�<��}O�����Y�c�U���Yתnbj�����0-��;c3�*id�pWI{���4�,h�8PkX%��J��-lJ-��F��zTu��I��n��86~��/J95�o�D����S��4�M/���l��T�\�Prevf�{{�tJ�/��7YK��yU�)�5ah�#�x���ݦ�\�(��)����|��K�<�s-ŵ��n�ԚdU�.��V4]�w�F�Iu�yܫQ,�w����Ԛ�`�:V~Ӱ�?�,K��T�X��`@\+�xș ��]�@vv3��4('.-}p��H���`,w}��^���1o�%��g�x�/s�?   ��
1e8f
�]K��8r�+���(�gfwg�X]ϩGw�Uӽ�KG��H� ��C*����Þ|���o�����[`	��z29	���/o `��Hۃ9n���O��E*%a8>�k�[W�@�+!W$�Ğ�m��������u(Voe��ۼѿV�\G_dqI�C��{��~x���t���I���_z%A��6dݵ���$��1}�-�RX���b��o_���r	�o��ͅ�8�O!4H�o�I����8*��J������*���)�+�)���`���>���æB���gDh��M,[6���p����϶P:��o�yuh�]Z��n�<�<�HW0�Wg^�r�J�AO��sV����U>hR��F���j��Φ9��0��@�#H���#��&)�(l�F�ȗ��Rt4�CS��?�~���{k��#'�|M:>��?���1���툔A���� 'C�) E�R�:��UpD��#it#Td�~��ԠKD4��Ն�S���(�|ȏ��;~Fk���L����N���B�]�`u^!?��)��(��=�]��=8ѝ�Ο
{!�,�9�P�~J��] �����J���M���������9��2y����v/�L�G%�ב�p�s*�]]bخ��h�+�~��~��Wmt�?Q�~��@������u��ӯg��*8~yurg����4>,���ٶ�z�̳�Ą��0kVwh�%-�-E�1N�����@s2�����ٻ��h��R�W�����������C�[G��D]���u��¡��d!�tqO�y��O(�ذ[AIkV��N���[^�V���uq>�i�H�*�
e�U,�S���Ҩ�
y
�ϣ�70�����`�Pz�[w�łӟ���s�t+ٔ�
�i?:xrD,Sg���9Y x(i���W�>n��}hZ�+��g�^�
�J�j�"�|#-�R��vZ ���0[�|]S0�\ʺ|�`{�"��9����a��{ ��'��M�<ڧ����v'�LKc�+���zo#�W��h"���6�y�~<�[3�LI��N�KV��AVd`]{h
\W�ڬ�Ծ��p$� 6����C��NVs�ʴl���X��&�w���y�=�;��ۑ�mM���~��C2B��,��d��ހ�{�!�U4���\�I$�rp�QV�A�Xz.]��
�[R��:��AC��X1�'���|�"�P�E#�[�����T��E���*���d�uL���:����>�>��V��7%�QYM�f�^
����9o��$�����&;�{�pm�a�_2x�J+�j ��,��m�Kn��t2A�?	��78n�
�L. #�"��v��~�/@�?c���k���?�-R��k��Қ�Q[�Ej�$]�V�.�X�P'~{ХD��.�$[�A�z�]g.~+��t�L��V������Ր���_&=�N�|e��'O��t�,�ֱw ��`e\W�yV��N(�
V#WY���2���I��V��������=k"���kϽ9�!����-���w~���:H�SW��>�UUMolF���_8��P�P\�UƱ����@��h��j��x�č����� Έp[�ɨ��0oj~�#R���R�q��CY�w/�;8.}UQ�b�m�>M�=��6�s�,��S��v��=Q�P�.?b�*�|lwF��B'hj��'�'��¾u<���j�����^�4���Jb�C�J4���%�t^O�G��<��~Ɠ�{�tuq�/><�z�W����b�]�AѢ�:�<���ܙ���V�,a���h-�ʒ,V[!��<�\�y���}�^��vl�mX�3�����p�?���e��v�9;b�^����Ag��
�+��9�C��]
+s�PT��f]X��%�`�"��ƅ��f0�i@߹+� �~"[�p��M��fB��(�q�sp��#V��2�g�YX/��D#�GO�v&
����Q9�-�xt�,�~��7?����$�K���Dp	��<&߫��f�x ]5M@�OG�ŜH���JbX}��!�ul��y�F�h���(�M���|��Ϧ_Ec,p��lc3� ydk2�%��gQ�Wl��LK�q�г;���+���?��Üwߑ�����ĭ��V�2q�^&)2��=����P�BdKZ��ϫ2��BCN2�U4F�ԛh�<A/�m�zp���͟������|gz�#�C}��\yב)Dt��8}@Ae�3��77ur��r#��������\d�$.�*�&���D��Um*�J9g���c8��7��q�]'�8]�E]��p�_�q�d���&i����W�A�}^�S+2���L��ū��Fčd���h_�yV�,�v�} ���y��Q`�̕���`(�`��p�d�
U�P��.�Ѱ[#zL���!�E�aT�\����`�y�"mCe��Q0Hs���DQ��!�$�'�ɺ�k4͈�H
r�l�h�&�i���-(!�Mk��TQ�"G��
5A��q;��b��^�֕�\)�_�]3#�]�9�_"²k�����'�4�'��BV�H�/����ޏBkCC�$��?����s�럀k��K}&%[�n��V��Z��G(�\��-�J�;%�Gg��wJ��Б��b���ʚ}M&�+�h:�"��E�L��p�Eۜ=�~�-M7<��y��*����ѱ�#r�@6�AO��_p$
��<�
�ֈ
��7�/Ux��xti��FUC�H?��4�P
����eZ�bA�<΂�蟕ӡ�
��X��u���d!�^@/p�a��E�G1Ԝ;��]��=~�h�]ԝ�ZA	��v�K���]<��>I���I9���.t��+�vס���h7Te��D��K�J��h�J��M���
���D��Oo�j���f�'�t�d���
m�sD��.�0�)l��4��N��@��r�F�i��~o�D�E���`m����rƆ�>�8��T��ؖ�*��y-wf ~��w?�Z�q+(E�ù7	�P����e��Na`�<�}����9���\^_H�f���.���x�̇��*�(ip��+ሱ,s��5�l�´
g�ms�6
d�uV�n4�#�R(1R�n���܂E͖/Be�Gլ�:^@	(7�cxa\6��j��<S���h_�>
*�+u�˟m������G�8#}����Pk8𾤋�f�������ue��� >�����-���y��t(�X!�Og�
��/�����5��y��y�j�B~TҔV�oP�D7����⯮4N0'�;t��Tݻ{�M�#^D�'�YA#i�b�"/�n����U91�ؖ���h��d>p�O35H��ە\�c��R0r5��Yh	�nKiݥ���������Q�({d-�b����S�0����P��^�� ��M�}g���_d>�p'w�K]��YR-mh��y� I�y�n�ϱ��+X��+DO<��Z`Y�+�E���ǋ=�y�[ \Ğ������8��4 u��8�1�Z�U��� }!�Hc�oM�����:��8Q�Z�c羖ޏD�1Q��ʽ�OI�}P���OnT�@'��7A��u[����3�^b#Q�w�1��F�ut̺ӹ����=B����>>�s��ǖ-p��qЅ�Y;�-wx&�[]��{��M����6�".j�'��
�S�0�j��c�*9i�Yv��6RHڀV�F`/���
��7�B��~V�_!��,�X9��i `T�d����4������0��B�9=K0��\ѥ�#��e��M=��K��5��� ���y��s��h��<��a":�3( �+�-aE��ѪF�~�k�
��h���:'��b*��@l��8ܐ��f�AQ	����0~'6�.`�6��Du�]�u�s���M��E"2T��K��7�߅�?:��*Q�Ґ�{֫,*�8(���Y-�{���ma�-����;8��5���(q�4m�wY���6��z��D��s��Qn}A�DF~���(�I�da���\�fj�%�7�fj��lsn�Xc�ˢ�lNS�T��Aǜ6��+����~�@Ӏ]l�rk�~[}u��?�7&�Y�T聢5ݔ��h�����T��:�6H =��)�ep{���/�ux5�5�!��l�TL��5�Q1�+�����6+�g���hp�[��6��W�8�/Zt��@�lAՖ|��~�<E��3 -��z2ϸ˫����q54��sn��"���Mg�d����ؽ�u8���ojzr̸S�l�z��9����A�f�C�������@��aOIl��j����8�������7���Y5���J�v>T��
��yƓ��5)�:�[7�:^w�	o�ڶ��N�
Y��i�=�Ͻd8�;�(��?#j 7�}� ��������tn] E��Z�%��uz���H�(U,jS�̑��J'�,Юt���=ǒ�� ߝ�1:�R����oY��]1clMjg�?   ��
1ffa
����k�����lE'���g���*�e�����t��ğ�M��R�g X�ܾe6�([��Wț��Ɖw���됯)����h��4��>8�
Wv6�bnsn��2��pWh�TCd�j�r�?�W1'h~�?�vˉD����H����&\�4t}ak�0�V��oX��!>���.meĠ2�mv~n�ǵ�T��DQ�.6�<e=�@��� @	C�ﰶ���)���eNx�h����\�F�m�$�
	��8H@UۢN�Υu�!+�䠝R��$"�o~GJ����P�#Wq�{	��y��� �j�|h�3���h����`)W�<��	�P,j��w!=�GW��J����&8��lXm��S�XsJ�tf�+�U��'N<!/0�:��/� U��W�p����
�XgZ��sU��]{�i�z��u��2k!`�
�\�Xh��j9<�(�"f.�v��������x�@Wn7ߺ")�y^�^☄ٍ,��v&5�xB�
��;m-|�ua���c܊��@
tD�'b��I��+q�Se9�g��Ǆ$�,�(�
O�ځ��cq��m^*ΰ���(�,����m;�c	1�'X~Þ�����t4�l�Z�\��R��rFv���x�8��CB��T��4b��O��\V��$I�Η3���-��:A#�=�d��2��0�VX�I��Q���+s�m��Xhݚ�c���UY.��qA#���7j`%_mwt�Y�>R�z9�gl�ڲBX���0fH5�B!EݏE�Fc,(��#J[�O�D�=	|x�9�?N�99�˅�Eժ/R�L����TU�Y��Be �sxm���`Ʌ�<�_F�m��=PN��:LP���.��1�}�K�/�͹Fw�H Gp[H%�K�#���}�'AgZ��7&q��%��Z3l=� �"�X8\a7q0���>��GҨ�ԹlEk��cj�|�6����ѹɛ��k�<�U�<!�l6X�L��z���We��tVV}l�D�E�H��B�[	�w�3��
�8�B�����sJ�4"�kR�1��c�O%�\+��~��`]��W�����W�_2z��\��>�m���9'���V��1Ac@�X�y�p����	d�d���i����Bh�ռ]�:�
&�%�&�)\]T�:���T���I�(AN	b͜9-E`�K
'8�� H���#{
��w����u�w� >U`�޻R*GH����t4�
��^���V�\��n'�F�A�[T��.潃��vI.�B��v��{�����&E@�0r�Q�+��A��=}-L�&����R3�O�azY�f��n^�Y�.�L������?-��"Kmѥk6|�-D�E��ǻ �n'=�V��Pt���r�Pv`��k.H�Bh2�
[�Y٣�h�%侥#���b�8��B��%M��Y��0~#!��ѝ�BNs!yh�:�;==;}��ݛӷo|���Y�E ݀
b� Iۗ�d�l�K���͗.ǳ�"\/Ԗ�N�`��I�q���]�(z��΃oWhK���[y�9lFj����^T�*<����T��l�;�K�/m��8���Ҩ�@��k8J�!�Ș!��n�T��%��T����(�r�|S�E�rÁz-��:}��@���}�j{D����D��Z
M,]�J����g6�_(�����<$�R�Ǻ*<�0[�i+0v��0֋L	eSj�ZAW���u��E.�^�2܈��פTkپ4e�:Wl��$Y�ҩHS����-�4��O�9�N`�G6W@h�SPI_e�e���%�1t�:���]�K��� ��
�aϥh�a	K[��/�H)r)R�]��K���L�b��4��b�Ԩ�Z� �*ry��_i��h}�c�㒓�)Gܕv;h�3�Y��W�*�h&���l�Yj�R��-U�oW��4y�?`.�Տ}��n�f%�5��a�c􍱛g Y��8�x��۰
�J,�Xdl
�C��ۧ^iXg���"@x逪Sz�dF���%]9�J���9aB���آ�)�N0O�Z+�Cd�����NT�6�p�N���P������,��ٜ?���;��l����^/,TE��[kXʚ�pp�IS�P8R�T�� lc\��|Ӥ��5.we+v/Z
�-��vH�0GRDESg%`��qx�^N��|aT�n��Jr ;Y=V�7�k��(B� [� �U7x�!� ��F�\q�P+�����N�`8w_P�����4=c,��9�"O'��w��pJp`t.+�R��;E�P	�ǳt��FX����4iQ8ߝ셑=�`Q�T7bV0{yŁ��H.��x���Ttn��Ĭ6�M}MЅy8�s�����b��©�ǘ�d	�2�l���]�E����p_��� -�Xz#孰1�|�����5A@��:l&G�w.��u�wE%�j�䀶N9z<!1������_��ɵ�j߻�7��"�=U�Y���D/�`��*
��H �ݚ���0�0���T)&R�V�@��
ZP�ct�Y`��f�)��^�K�<����:����4�ݭl��h���ת��|��sBy	�Kl}�\`�}
	eD�EU��(�������H ���6>�"WhFS�!��q�r��̆��=E���A�I��r"5��Wwf#{�j��a)k�;�y��\,�;/�ݩ"i���Z�RG��0�dP�o�q���%6AG$l�
��Ƒ����P���>U Q����/�n�|�,jP�T�5k�ś'<�^-�-�����s^F �îɵm��������2��TK2�!��qٳ��)�JBm���� �D�Z�;�-����$w^��F\���Q���aQ/D*��*�%��TY�(��Pr� ]ֶ�E^Ƣm�<��j��#�����UtK��6!��%u��-�&���y���x����C����N�����qB�-�tΥ��g<#L�8�� <��ˎrߦ�<4i`�W��;�|�*V��0���Ȃts/@�kthx��� �k��EĚʎl�zOA����#[�X�pQ�1���!�����^���TD��!Y��n]I��z*�Ej�<�"�W�'&ŭD�	F��<АM�ݶ��/��MA�).�B����F��vK�;�
��+s�
<U���&nմ<�(���ͭhЍ�<2��L���sW�ɸ�[����xB�ЛK�
�0J1$]b;�澑�`&�n:Fش��	��=�*]a�1��:���n�0O)Qk��a�����n,:�b$��L�-�گ�Qk��H��U��U�8@��z�^�,'�	��k@�4
�aU鶠՗��tޗ���D�30��!�Ѳꂋ#�)���@(t.�AK��w����5��hE�y[cW3q,Rd-�x��3���wf�1U%���c��
\�K���q�H�h�OJG(3쓞�q�x����h�d�7"�ʹ�E_�j�O`5eAz޾ jdZZ��.8Y,�C.���@��x\��x�Q.��I/׺R76��<$JOUf��=Ⱥ.���m�{�*�9��N?�5"�I�#0C��C"	��(�"��q�`�c$��D.,�y
b�4���""
�BåP)�P$u#�'O�<5J�:z�Q]�=�<E�h�����ɵ2?����(�װ��f(C����:M&aIRjʹy���D��o�D����[����j�7�`�2�on͕���Çٽ�8ߞ��T�Buffi�ސ1z�@�$�OI������G�k��T��d'�I�NI�C�����x#�^=`�X�5R6�f?���!�kF��J,3�a���栺x�Kܐ=D;P���H�|��$
�cF3�A\'��������x<���y���$ZmX:+�Qb���������/1Ǚ���ZKabFT.ॢ\���;k��_�B��%�5R��J�_I1�F���3B�9����>y&��ik��F��x�W��??����x�{@HyO��oN���
���r⟔8�� p�&^�k\���6�\A�3���f�0�'���V�����I?��� {�?s4o�٪�ڍsq�k:�7�l�#B���=%c4��w������S���H�>!�c?�Y��Y&If�5�e���+FNq����:������J�*~ҫ,<��'�o@c��	�_��Y��*�~�\qF�'���t
O�bD$�5�$�������n��D0!"��������uڒ1�2ʹR)������ٸ�&�	��&��g����A
����3�����d����rL!!4A��)�$���s�RY�|&::�����^kw�D���/��P�g^:7*R�`��A���a��QN%��2]ٰ���.��+����r�����N��[R\Q]?�0�3��j�@�V`~� ���ٜ��(W��ap��@n
�}A� �����ÜM���4����Ѡֹ3���HM�^�s�H��
�yŝ���pn��a�H���Ra��t}q��\��U�-ṛ�%/ӘQ�I\�߹�g�溴/D�� �eD�A��	�\�\h��R
o^�=�����+���l�/]�����ٍ\kKC.�}�c��?�;����ȉ'*��,��f��m2�T#<�D[���'�����~�ŏ�jq��9��o� $��,���ޜ�ʆ�-P52ƛح�|�k[
9e5
K���=��c#�
i8���
��C��a�)�q�)�
��(H���g`$�I��1V���H4Nư��)�hE޺Jy�������jA[���4�r�(8����a�~	�)zI,��_<�ww���P�U�+�a�%Ɉ��S,�������?{>��o���_���:0�ά����*y.$i��W˳���+��%ڳ4fN�~EM4.�ua�LͲP�ͼiV���U����8�d��VB)
q��\�� �s�#�A�=DCzAҹ�J(
�W=ۯ�g;���8:�a��������z�l<��+�fu�������`z�鎖���6�x�/B�o<�V�gs�?hv�m}i<pӅ`�L���u����2��
�h���b���sNR�T������6
6��ܪ�O�ju�{cvw�U�&�V�xs��¾������?y�YƼ�Ra�'�/��ſ��lMW�@�t�Lt��mq;�&�1���S�X"����3/^���O&}���s�������1���L���7�1��v�_�5t3�8JY�Qٛ6&s�o���Kx����G�1"ѭ{;����3{��mo]�G�[�=��Zӣ�=�̎��.f��u�����4�{U���${Ա[0P%N ��]��	u�1wn�6�+h
^�j�!�1?,�km����x�]io�Kt����������n��2*����g��$O��;��{�{������n��vp)�}
X��t��7~ykl���|��Y�`Z��3|�d����j@�"�N�o��&p�G����2�����ߣR�<DZr�
�S�P��x��g����Y6�g̙�jhL@�E%�˧�����ݪ��&vXv���6,���  �� E���5~ 
0

```
#### Expected result

console will print a html just like:

```html
GET / HTTP/1.1
Host: www.wireshark.org:8080
User-Agent: curl/7.46.0
Accept: */*
Connection: close
Accept-Encoding: chunked, gzip

HTTP/1.1 200 OK
Server: cloudflare-nginx
Date: Wed, 06 Jan 2016 20:42:10 GMT
Content-Type: text/html
Transfer-Encoding: chunked
Connection: close
Set-Cookie: __cfduid=d8d37b52eaa3137bdfd7fd67a4ffc8a7a1452112929; expires=Thu, 05-Jan-17 20:42:09 GMT; path=/; domain=.wireshark.org; HttpOnly
X-Frame-Options: SAMEORIGIN
Strict-Transport-Security: max-age=31536000;
X-Slogan: It's a great product with a great story to tell. I'm pumped!
X-Mod-Pagespeed: 1.9.32.11-7550
Vary: Accept-Encoding
Cache-control: max-age=0, no-cache, no-store
X-Slogan: Go deep.
CF-RAY: 260a3f709d7b0761-AMS
Content-Encoding: gzip


<!DOCTYPE html>
<html>
<head>
```

#### Related resources

[http-chunked-gzip.zip](https://github.com/secdev/scapy/files/5011931/http-chunked-gzip.zip)

**Comment by gpotter2** [comment]
Thanks for the report, this should be fixed in https://github.com/secdev/scapy/pull/2745.
Feel free to try it out

**Comment by Qnner** [comment]
> Thanks for the report, this should be fixed in #2745.
> Feel free to try it out

Thanks for your help. I'v tried your code, but there is an error in sessions.py in line 82
by the way, is  the 'print' line 84 correct?

```python
C:\Python27\python.exe D:/Sniffer/SniffTraffic/Utils/test_response.py
Traceback (most recent call last):
  File "D:/Sniffer/SniffTraffic/Utils/test_response.py", line 48, in <module>
    sniff(offline=r'C:\Users\test1\Downloads\http-chunked-gzip.pcap', prn=packet_parse, session=TCPSession)
  File "C:\Python27\lib\site-packages\scapy\sendrecv.py", line 972, in sniff
    sniffer._run(*args, **kwargs)
  File "C:\Python27\lib\site-packages\scapy\sendrecv.py", line 925, in _run
    session.on_packet_received(p)
  File "C:\Python27\lib\site-packages\scapy\sessions.py", line 527, in on_packet_received
    DefaultSession.on_packet_received(self, pkt)
  File "C:\Python27\lib\site-packages\scapy\sessions.py", line 318, in on_packet_received
    result = self.prn(pkt)
TypeError: 'bool' object is not callable
```

**Comment by gpotter2** [comment]
Aren't you passing it a wrong object ? (bool in `prn`) . Check your types
Yeah the print was a mistake. I removed it
